tests/sentry/utils/performance_issues/test_performance_detection.py

```diff
@@ -922,6 +932,54 @@ def test_detects_slow_span_in_solved_n_plus_one_query(self):
             ]
         )
 
+    def test_detects_consecutive_db_spans(self):
+        SECOND = 1000
+        span_duration = 5 * SECOND
+        spans = [
+            create_span("db", span_duration, "SELECT `customer`.`id` FROM `customers`"),
+            create_span("db", span_duration, "SELECT `order`.`id` FROM `books_author`"),
+            create_span("db", span_duration, "SELECT `product`.`id` FROM `products`"),
+        ]
+        spans = list(
+            map(
+                lambda span: modify_span_start(span, span_duration * spans.index(span)), spans
+            )  # ensures spans don't overlap
+        )
+        consecutive_db_event = create_event(
+            spans,
+            "a" * 16,
+        )
+
+        sdk_span_mock = Mock()
+
+        _detect_performance_problems(consecutive_db_event, sdk_span_mock)
+        sdk_span_mock.containing_transaction.set_tag.assert_has_calls(
+            [
+                call("_pi_all_issue_count", 4),
+                call("_pi_sdk_name", "sentry.python"),
+                call("_pi_transaction", "aaaaaaaaaaaaaaaa"),
+                call("_pi_consecutive_db_fp", "1-GroupType.PERORMANCE_CONSECUTIVE_DB_OP"),
+                call(
+                    "_pi_consecutive_db",
+                    {
+                        "trace_id": "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+                        "parent_span_id": "aaaaaaaaaaaaaaaa",
+                        "span_id": "bbbbbbbbbbbbbbbb",
+                        "start_timestamp": 0.0,
+                        "timestamp": 5.0,
+                        "same_process_as_parent": True,
+                        "op": "db",
+                        "description": "SELECT `customer`.`id` FROM `customers`",
+                        "fingerprint": None,
+                        "tags": None,
+                        "data": None,
+                        "hash": "",
+                    },
+                ),
+                call("_pi_slow_span", "bbbbbbbbbbbbbbbb"),
+            ]
+        )
+
```

Comment:
I would also suggest test cases for the two other main criteria for this detector: that it doesn't detect when spans overlap, and it doesn't detect when spans after the first contain a `WHERE` clause. Because we're only doing detection for now I wouldn't necessarily block this PR on it though.

Reply:
Yeah, only detection for now! Definetely have plan to improve testing